File: QueryOptimizer.Models/Application/OptimizationCandidateResult.cs
using QueryOptimizer.Shared.Common.Models.Metrics;
using System;
using System.Collections.Generic;
using System.Text;

namespace QueryOptimizer.Models.Application
{
    public class OptimizationCandidateResult
    {
        public int CandidateId { get; set; }
        public string RuleCode { get; set; } = default!;
        public string CandidateSql { get; set; } = default!;
        public string? Description { get; set; }

        public bool WasTested { get; set; }

        public bool IsBest { get; set; }

        public QueryPerformanceMetrics? Metrics { get; set; }

        public double? ImprovementPercent { get; set; }
    }
}

File: QueryOptimizer.API/app/query-optimizer-front/src/app/models/optimizationCandidateResult.ts
import { QueryPerformanceMetrics } from "./queryPerformanceMetrics";

export interface OptimizationCandidateResult {
  candidateId?: number | null;

  ruleCode: string;

  candidateSql: string;

  description?: string | null;

  wasTested: boolean;

  isBest: boolean;

  improvementPercent?: number | null;

  metrics?: QueryPerformanceMetrics | null;
}